### PR TITLE
Adding torch examples

### PR DESCRIPTION
**Pull request by ncguilbeault**
This PR introduces new examples for the `Bonsai.ML.Torch` package in https://github.com/bonsai-rx/machinelearning/pull/48. A list of examples and brief summaries are provided below.

# Examples

## Linear Regression

In this example, the `Bonsai.ML.Torch` package is used to perform simple linear regression using the ordinary least squares method on simulated data.

## Neural Nets - Pretrained Model

This example shows how to run inference using a pretrained convolutional neural network to solve a typical image classification problem. The network is trained to categorize clothing items in the [fashion-mnist](https://github.com/zalandoresearch/fashion-mnist) dataset and perform online inference of new images that were not included in the training dataset. In this particular example, the pretrained weights of the model are loaded from disk.

## Neural Nets - Torch Hub

This example shows how a PyTorch model can be downloaded from the [PyTorch hub](https://pytorch.org/hub/) and used to run online inference in Bonsai. For this particular example, the [MiDaS](https://pytorch.org/hub/intelisl_midas_v2/) model is used to infer the relative inverse depth of images streamed from a camera.

## Neural Nets - Training Online

This example shows how a convolutional neural network can be trained online to solve an image classification problem. It demonstrates a typical training procedure that involves performing SGD on the weights of the network as new images and labels are passed to the network. After a certain amount of training, the model can be switch to online inference only mode.

## Running On Mono

This example demonstrates how to run the package on Mono.  This currently requires a workaround to the DLL import which explicitly throws an error on non-windows OS's running .NET Framework (which is the case for Mono). See this issue: https://github.com/dotnet/TorchSharp/issues/689.

### CODE REVIEW

File: examples/Torch/NeuralNetsTorchHub/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Running Inference using a Torch Hub Model
+
+In the following example, you will see how to run inference using a model downloaded directly from [torch hub](https://pytorch.org/hub/). The model we will be using is called the `MiDaS` model, which takes 2D images in RGB space and converts them into 1D images representing depth - more information about this model can be found [here](https://pytorch.org/hub/intelisl_midas_v2/).
+
+### Python Script
+
+This example contains a python script called `torchhub_download.py`. To run this, you need to have a python environment with the pytorch package installed. To install pytorch in python, you can follow the [instructions on the pytorch website](https://pytorch.org/get-started/locally/). Afterwards, you need to change to the example directory and run:
+
+```cmd
+python torchhub_download.py
```
Comment (glopesdev) [inline examples/Torch/NeuralNetsTorchHub/README.md:10]:
```suggestion
uv run torchhub_download.py
```

Consider replacing instructions with a [uv](https://docs.astral.sh/uv/getting-started/installation/)-ready environment so we can make sure all dependencies get installed and the script is run with the above one-liner.

File: examples/Torch/NeuralNetsTorchHub/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+```
+
+This will download the model from torch hub and save it to a `.pt` file. The file will be saved inside of the `models` directory.
+
```
Comment (glopesdev) [inline examples/Torch/NeuralNetsTorchHub/README.md:14]: Consider reinforcing the instructions for libtorch deployment in case someone is parachuting into this example.

File: examples/Torch/NeuralNetsTrainedOnline/Extensions/FashionMnistLoader.cs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+using System.ComponentModel;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using System.IO.Compression;
+using System.Reactive.Linq;
+using OpenCV.Net;
+
+[Combinator]
+[Description("")]
```
Comment (glopesdev) [inline examples/Torch/NeuralNetsTrainedOnline/Extensions/FashionMnistLoader.cs:12]: Maybe add a description here. Just a note in the future we might consider how to handle this kind of shared extensions, but for now is fine.

File: examples/Torch/NeuralNetsTrainedOnline/NeuralNetsTrainedOnline.bonsai [hunk 4]
```diff
@@ -0,0 +1,887 @@
...
+                 xmlns="https://bonsai-rx.org/2018/workflow">
+  <Workflow>
+    <Nodes>
+      <Expression xsi:type="GroupWorkflow">
+        <Name>LoadData</Name>
+        <Workflow>
+          <Nodes>
+            <Expression xsi:type="Combinator">
+              <Combinator xsi:type="p1:FashionMnistLoader">
+                <p1:Path>../../../datasets</p1:Path>
```
Comment (glopesdev) [inline examples/Torch/NeuralNetsTrainedOnline/NeuralNetsTrainedOnline.bonsai:22]:
```suggestion
                <p1:Path>../../../datasets/fashion-mnist</p1:Path>
```

File: examples/Torch/NeuralNetsTrainedOnline/NeuralNetsTrainedOnline.sln
Comment (glopesdev) [inline examples/Torch/NeuralNetsTrainedOnline/NeuralNetsTrainedOnline.sln:1]: Maybe consider removing this file.

File: examples/Torch/NeuralNetsTrainedOnline/README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# Image Classification using a Neural Network Trained Online
+
+In the following example, you will see how to train a neural network online to perform image classification using the fashion MNIST dataset.
+
+### Dataset
+
+The dataset used in this example can be obtained by going to [this url](https://github.com/zalandoresearch/fashion-mnist?tab=readme-ov-file#get-the-data). This example uses both the training and testing datasets, so download all 4 of the urls. The files need to be converted into images and text files and placed into the `datasets/fashion-mnist` folder. You can convert the compressed files into the correct formats using [this tool](https://github.com/ncguilbeault/fashion-mnist-dataset-export).
```
Comment (glopesdev) [inline examples/Torch/NeuralNetsTrainedOnline/README.md:7]: Maybe these instructions need updating, but the target folder is correct.

**Review by glopesdev — COMMENTED** [review]
Everything looks good, just a few suggestions to smooth out the process. Created a branch `dev/torch-examples-torchhub` with a suggestion to improve python bootstrapping using [uv](https://docs.astral.sh/uv/getting-started/installation/).

File: examples/Torch/NeuralNetsTrainedOnline/README.md (hunk 5 above)
Comment (ncguilbeault) [reply examples/Torch/NeuralNetsTrainedOnline/README.md]: Updated

File: examples/Torch/NeuralNetsTrainedOnline/NeuralNetsTrainedOnline.sln
Comment (ncguilbeault) [reply examples/Torch/NeuralNetsTrainedOnline/NeuralNetsTrainedOnline.sln:1]: Removed

File: examples/Torch/NeuralNetsTrainedOnline/NeuralNetsTrainedOnline.bonsai (hunk 4 above)
Comment (ncguilbeault) [reply examples/Torch/NeuralNetsTrainedOnline/NeuralNetsTrainedOnline.bonsai]: Updated

File: examples/Torch/NeuralNetsTrainedOnline/Extensions/FashionMnistLoader.cs (hunk 3 above)
Comment (ncguilbeault) [reply examples/Torch/NeuralNetsTrainedOnline/Extensions/FashionMnistLoader.cs]: Added

File: examples/Torch/NeuralNetsPretrainedModel/README.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# Image Classification using a Neural Network with Pretrained Model Weights
+
+In the following example, you will see how to use a pretrained torch model to perform online image classification using the fashion MNIST dataset.
+
+### Dataset
+
+The dataset used in this example can be obtained by going to [this url](https://github.com/zalandoresearch/fashion-mnist?tab=readme-ov-file#get-the-data). Only the files `t10k-images-idx3-ubyte.gz` and `t10k-labels-idx1-ubyte.gz` are needed for this example. The workflow expects the datasets to be placed into the `datasets` folder.
```
[the remark this answers is not in the document]
Comment (ncguilbeault) [reply examples/Torch/NeuralNetsPretrainedModel/README.md]: Updated the instructions

File: examples/Torch/NeuralNetsPretrainedModel/Extensions/FashionMnistLoader.cs [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+using System.ComponentModel;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using System.IO.Compression;
+using System.Reactive.Linq;
+using OpenCV.Net;
+
+[Combinator]
+[Description("")]
```
[the remark this answers is not in the document]
Comment (ncguilbeault) [reply examples/Torch/NeuralNetsPretrainedModel/Extensions/FashionMnistLoader.cs]: Added

File: examples/Torch/NeuralNetsPretrainedModel/NeuralNetsPretrainedModel.bonsai [hunk 8]
```diff
@@ -0,0 +1,516 @@
...
+                 xmlns="https://bonsai-rx.org/2018/workflow">
+  <Workflow>
+    <Nodes>
+      <Expression xsi:type="GroupWorkflow">
+        <Name>LoadData</Name>
+        <Workflow>
+          <Nodes>
+            <Expression xsi:type="Combinator">
+              <Combinator xsi:type="p1:FashionMnistLoader">
+                <p1:Path>../../../datasets</p1:Path>
```
[the remark this answers is not in the document]
Comment (ncguilbeault) [reply examples/Torch/NeuralNetsPretrainedModel/NeuralNetsPretrainedModel.bonsai]: Updated

File: examples/Torch/NeuralNetsTorchHub/README.md (hunk 2 above)
Comment (ncguilbeault) [reply examples/Torch/NeuralNetsTorchHub/README.md:17]: Improved instructions for uv and torch

**Review by ncguilbeault — COMMENTED** [review]
Thanks for the feedback. I think I've addressed all of the comments from the previous review. Let me know what you think!

**Review by glopesdev — APPROVED** [review]
